File: de.lmu.ifi.dbs.knowing.core.swt/src/de/lmu/ifi/dbs/knowing/core/swt/internal/Activator.java
package de.lmu.ifi.dbs.knowing.core.swt.internal;

import org.eclipse.ui.plugin.AbstractUIPlugin;
import org.osgi.framework.BundleContext;
import org.osgi.framework.ServiceRegistration;

import de.lmu.ifi.dbs.knowing.core.factory.IPresenterFactory;
import de.lmu.ifi.dbs.knowing.core.swt.factory.MultiTablePresenterFactory;
import de.lmu.ifi.dbs.knowing.core.swt.factory.TablePresenterFactory;

/**
 * The activator class controls the plug-in life cycle
 */
public class Activator extends AbstractUIPlugin {

	// The plug-in ID
	public static final String PLUGIN_ID = "de.lmu.ifi.dbs.knowing.core.swt"; //$NON-NLS-1$

	// The shared instance
	private static Activator plugin;
	
	//
	private ServiceRegistration tablePresenterService;
	private ServiceRegistration multiTablePresenterService;
	
	/**
	 * The constructor
	 */
	public Activator() {
	}

	/*
	 * (non-Javadoc)
	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#start(org.osgi.framework.BundleContext)
	 */
	public void start(BundleContext context) throws Exception {
		super.start(context);
		plugin = this;
		
		tablePresenterService = context.registerService(IPresenterFactory.class.getName(), new TablePresenterFactory(),
				null);
		multiTablePresenterService = context.registerService(IPresenterFactory.class.getName(), new MultiTablePresenterFactory(),
				null);
	}

	/*
	 * (non-Javadoc)
	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#stop(org.osgi.framework.BundleContext)
	 */
	public void stop(BundleContext context) throws Exception {
		plugin = null;
		super.stop(context);
		tablePresenterService.unregister();
		multiTablePresenterService.unregister();
	}

	/**
	 * Returns the shared instance
	 *
	 * @return the shared instance
	 */
	public static Activator getDefault() {
		return plugin;
	}

}

File: de.lmu.ifi.dbs.knowing.core.swt/src/de/lmu/ifi/dbs/knowing/core/swt/internal/Activator.scala
/**
 *
 */
package de.lmu.ifi.dbs.knowing.core.swt.internal

import java.io.File
import java.io.FileInputStream
import java.io.IOException
import java.io.InputStreamReader
import java.io.LineNumberReader
import java.io.PrintWriter
import org.osgi.framework.BundleContext
import org.osgi.framework.ServiceRegistration
import de.lmu.ifi.dbs.knowing.core.factory.TFactory
import de.lmu.ifi.dbs.knowing.core.util.OSGIUtil
import de.lmu.ifi.dbs.knowing.core.swt._
import org.osgi.util.tracker.ServiceTracker
import org.osgi.framework.BundleActivator
import de.lmu.ifi.dbs.knowing.core.service._


/**
 * @author Nepomuk Seiler
 * @version 0.1
 * @since 22.04.2011
 *
 */
class Activator extends BundleActivator {

	private var osgi: OSGIUtil = _

	/**
	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#start(org.osgi.framework.BundleContext)
	 */
	override def start(context: BundleContext) = {
		Activator.plugin = this

		osgi = new OSGIUtil(context)
		osgi.registerPresenter(new TablePresenterFactory)
	}

	/**
	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#stop(org.osgi.framework.BundleContext)
	 */
	override def stop(context: BundleContext) = {
		osgi.deregisterAll
		osgi = null
		Activator.plugin = null
	}

}

object Activator {
	// The plug-in ID
	val PLUGIN_ID = "de.lmu.ifi.dbs.knowing.core.swt"; //$NON-NLS-1$

	// The shared instance
	private var plugin: Activator = _

	/**
	 * Returns the shared instance
	 *
	 * @return the shared instance
	 */
	def getDefault: Activator = plugin
}
